pkg/server/plugin/datastore/datastore.go

```diff
@@ -40,9 +38,9 @@ type DataStore interface {
 	UpdateAttestedNode(context.Context, *common.AttestedNode, *common.AttestedNodeMask) (*common.AttestedNode, error)
 
 	// Node selectors
-	GetNodeSelectors(context.Context, *GetNodeSelectorsRequest) (*GetNodeSelectorsResponse, error)
+	GetNodeSelectors(ctx context.Context, spiffeID string, tolerateStale bool) (*NodeSelectors, error)
```

Comment:
I feel that whether or not we are tolerating stale selectors should be very apparent from the call site. This was explicit before with the `TolerateStale` field on the request struct, but is much less apparent with a bool parameter.

I wonder then if we shouldn't introduce a typed constant here that we can use instead of the bool, maybe something like:
```
type FreshnessTolerance int

const (
    Fresh Freshness = iota
    TolerateStale
)
```

and then at the call site, instead of:
```
ds.GetNodeSelectors(ctx, spiffeID, true)
```
you'd have:
```
ds.GetNodeSelectors(ctx, spiffeID, datastore.TolerateStale)
```

I don't really like the names I just picked for the example :laughing: so please feel free to iterate on them. The explicitness is what is important to me.

Reply:
I like this idea; just pushed a proposed implementation that feels like it matches the implementation language in the sql plugin.